File: rust/src/models/customer_list_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{customer::Customer, pagination_response::PaginationResponse};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct CustomerListResponse {
    pub data: Vec<Customer>,

    pub pagination_meta: PaginationResponse,
}

impl CustomerListResponse {
    pub fn new(data: Vec<Customer>, pagination_meta: PaginationResponse) -> Self {
        Self {
            data,
            pagination_meta,
        }
    }
}

File: typescript/src/models/customerListResponse.ts
// this file is @generated
import { type Customer, CustomerSerializer } from "./customer";
import {
  type PaginationResponse,
  PaginationResponseSerializer,
} from "./paginationResponse";

export interface CustomerListResponse {
  data: Customer[];

  paginationMeta: PaginationResponse;
}

export const CustomerListResponseSerializer = {
  _fromJsonObject(object: any): CustomerListResponse {
    return {
      data: object["data"].map((item: any) => CustomerSerializer._fromJsonObject(item)),
      paginationMeta: PaginationResponseSerializer._fromJsonObject(
        object["pagination_meta"]
      ),
    };
  },

  _toJsonObject(self: CustomerListResponse): any {
    return {
      data: self.data.map((item: any) => CustomerSerializer._toJsonObject(item)),
      pagination_meta: PaginationResponseSerializer._toJsonObject(self.paginationMeta),
    };
  },
};
